processing/src/main/java/io/druid/query/groupby/epinephelinae/RowBasedGrouperHelper.java

```diff
@@ -651,6 +657,38 @@ public Object get()
     }
 
     @Override
+    public NumericColumnSelector makeMathExpressionSelector(String expression)
+    {
+      final Expr parsed = Parser.parse(expression);
+
+        final Set<String> required = Sets.newHashSet(Parser.findRequiredBindings(parsed));
+        final Map<String, Supplier<Number>> values = Maps.newHashMapWithExpectedSize(required.size());
+
+        for (final String columnName : required) {
+          values.put(
+              columnName, new Supplier<Number>()
+              {
+                @Override
+                public Number get()
+                {
+                  return row.get().getFloatMetric(columnName);
```

Comment:
Hmm, this could probably do something like,

```java
Object val = row.getRaw(columnName);
return val instanceof Number ? val : row.get().getFloatMetric(columnName);
```

Most of the time, val should be an instanceof number. The getFloatMetric should only trigger if it's a string we're trying to cast to a number.